impl/src/main/java/io/kroxylicious/testing/kafka/testcontainers/FloatingTagPullPolicy.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.testing.kafka.testcontainers;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.regex.Pattern;
+
+import org.testcontainers.images.AbstractImagePullPolicy;
+import org.testcontainers.images.ImageData;
+import org.testcontainers.images.ImagePullPolicy;
+import org.testcontainers.images.PullPolicy;
+import org.testcontainers.utility.DockerImageName;
+
+import io.kroxylicious.testing.kafka.common.Version;
+
+public class FloatingTagPullPolicy extends AbstractImagePullPolicy implements ImagePullPolicy {
```

Comment:
Why not make this a KroxyliciousPullPolicy that knows about the special rules for latest etc and otherwise just falls back to PullPolicy.defaultPolicy() for all other tags.  That way TestcontainerKafkaCluster can just use the policy unconditionally.

Reply:
Not sure I follow. I think this could be applied unconditionally, it's not currently however.

Extending the abstract means we are only invoked if there is already an image in the cache.

Reply:
> I think this could be applied unconditionally, it's not currently however.

that's what I was trying to say.